haystack/components/evaluators/llm_evaluator.py

```diff
@@ -53,34 +53,87 @@ def __init__(
             The inputs that the component expects as incoming connections and that it evaluates.
             Each input is a tuple of an input name and input type. Input types must be lists.
         :param outputs:
-            Optional output names of the evaluation results. They correspond to keys in the output dictionary.
+            Output names of the evaluation results. They correspond to keys in the output dictionary.
             The default is a single key "score".
+        :param examples:
+            Few-shot examples conforming to the expected input and output format as defined in the `inputs` and
+             `outputs` parameters.
+            Each example is a dictionary with keys "inputs" and "outputs"
+            They contain the input and output as dictionaries respectively.
         :param api:
             The API to use for calling an LLM through a Generator.
             Supported APIs: "openai".
         :param api_key:
             The API key.
-        :param examples:
-            Optional few-shot examples conforming to the expected input and output format as defined in the `inputs` and
-             `outputs` parameters.
-            Each example is a dictionary with keys "inputs" and "outputs"
-            They contain the input and output as dictionaries respectively.
+
         """
+        self.validate_init_parameters(inputs, outputs, examples)
+
         self.instructions = instructions
         self.inputs = inputs
-        self.outputs = outputs or ["score"]
+        self.outputs = outputs
+        self.examples = examples
         self.api = api
         self.api_key = api_key
-        self.examples = examples
-        expected_inputs = dict(inputs)
+
         if api == "openai":
             self.generator = OpenAIGenerator(api_key=api_key)
         else:
             raise ValueError(f"Unsupported API: {api}")
 
-        component.set_input_types(self, **expected_inputs)
+        template = self.prepare_template()
+        self.builder = PromptBuilder(template=template)
+
+        component.set_input_types(self, **dict(inputs))
+
+    def validate_init_parameters(
+        self, inputs: List[Tuple[str, Type[List]]], outputs: List[str], examples: List[Dict[str, Any]]
+    ):
+        """
+        Validate the init parameters.
+
+        :param inputs:
+            The inputs to validate.
+        :param outputs:
+            The outputs to validate.
+        :param examples:
+            The examples to validate.
+
+        :raises ValueError:
+            If the inputs are not a list of tuples with a string and a type of list.
```

Comment:
Let's also enforce that the inputs are lists of strings.

Reply:
Sorry, I don't understand. Right now, `inputs` are `List[Tuple[str, Type[List]]]`. Do you mean we should limit it to `List[str]` so that users cannot use any other type than `str` for the inputs?

Reply:
After internal discussion, we've decided not to restrict the types of the inputs further as it complicates the code unnecessarily.